stdlib/test/builtin/test_uint.mojo

```diff
@@ -12,7 +12,136 @@
 # ===----------------------------------------------------------------------=== #
 # RUN: %mojo %s
 
-from testing import assert_equal
+from testing import assert_equal, assert_true, assert_false
+
+# TODO: compute those with ** when IntLiteral supports __pow__()
+alias ABOVE_32_BITS = 4398046511104  # 2**42
+alias CLOSE_TO_UINTMAX = 18446744073709551605  # 2**64 - 10
+alias UINTMAX = 18446744073709551615  # 2**64 - 1
+
+
+def test_uint_lower_than():
+    assert_true(UInt(1) < UInt(2))
```

Comment:
```suggestion
    assert_true(UInt.__lt__(UInt(1), UInt(2))
```

**Request** Here and below, could we change these tests to name the dunder method they're testing explicitly?

Writing the test using the dunder method explicitly means we can be very confident future regressions won't be masked by the presence of a "r" variant of these methods, or implicit conversion shennigans.

This is consistent with work @laszlokindrat has been doing very recently to change the existing `Int` tests to use this style.